webapp/src/Controller/API/GroupController.php

```diff
@@ -89,14 +87,23 @@ public function singleAction(Request $request, string $id): Response
         description: 'Returns the added group',
         content: new Model(type: TeamCategory::class)
     )]
-    public function addAction(Request $request, ImportExportService $importExport): Response
-    {
+    public function addAction(
+        #[MapRequestPayload(validationFailedStatusCode: Response::HTTP_BAD_REQUEST)]
+        TeamCategoryPost $teamCategoryPost,
+        Request $request,
+        ImportExportService $importExport
+    ): Response {
         $saved = [];
-        $postedData = $request->request->all();
-        if (array_key_exists('id', $postedData)) {
-            throw new BadRequestHttpException("Cannot add group with ID");
-        }
-        $importExport->importGroupsJson([$postedData], $message, $saved);
+        $importExport->importGroupsJson([
```

Comment:
Should we add a `//TODO` to change the importGroupsJson to do the unpacking?

Reply:
Yeah was thinking the same